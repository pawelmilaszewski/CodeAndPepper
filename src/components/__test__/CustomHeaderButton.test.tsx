import React from 'react';
import {CustomHeaderButton} from '../index';
import {render} from '@testing-library/react-native';

test('render custom header button', () => {
  const func = () => jest.fn();
  const tree = render(
    <CustomHeaderButton text={'People'} onPress={func} selected={true} />,
  );

  const {toJSON, getByTestId} = tree;

  const result = getByTestId('text');
  expect(result.props.children).toBe('People');

  const button = getByTestId('button');
  expect(button).toBeTruthy();

  expect(toJSON()).toMatchSnapshot();
});
