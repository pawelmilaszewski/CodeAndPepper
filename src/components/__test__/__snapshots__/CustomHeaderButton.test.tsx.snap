// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render custom header button 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#7CFC00",
      "borderRadius": 10,
      "height": 54,
      "justifyContent": "center",
      "marginTop": 32,
      "opacity": 1,
      "width": 108,
    }
  }
  testID="button"
>
  <Text
    style={
      Object {
        "fontSize": 20,
        "letterSpacing": 1,
        "lineHeight": 24,
      }
    }
    testID="text"
  >
    People
  </Text>
</View>
`;
